token/tokens.go

```diff
@@ -0,0 +1,156 @@
+package token
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/pkg/errors"
+)
+
+type ClusterTokenService interface {
+	Get(config *configuration.Data) error
+}
+
+type ClusterTokenClient struct {
+	AuthServiceAccountToken string
+}
+
+func (c *ClusterTokenClient) Get(config *configuration.Data) error {
+	payload := strings.NewReader("grant_type=" + config.GetAuthGrantType() + "&client_id=" +
+		config.GetAuthClientID() + "&client_secret=" + config.GetClientSecret())
+
+	req, err := http.NewRequest("POST", config.GetAuthURL()+"/api/token", payload)
```

Comment:
Instead of using a regular http client it's better to use Auth client. See an example here - https://github.com/fabric8-services/fabric8-wit/blob/master/auth/resource.go#L55
You will need to generate the auth client package in make for that - https://github.com/fabric8-services/fabric8-wit/blob/master/Makefile#L248

Reply:
as per shoubhik's suggestion, I will park this for a little while and make those changes as soon as it is usable or working!

Reply:
but yes noted

Reply:
Your code is in better shape now, please go ahead and start using the Auth client.

Reply:
this is done!